nixos/modules/services/web-apps/hledger-web.nix

```diff
@@ -50,28 +74,60 @@ in {
         Base URL, when sharing over a network.
       '';
     };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      example = [ "--forecast" ];
+      description = ''
+        Extra command line arguments to pass to hledger-web.
+      '';
+    };
+
   };
 
   config = mkIf cfg.enable {
-    systemd.services.hledger-web = {
+
+    users.extraUsers.hledger = {
+      name = "hledger";
+      group = "hledger";
+      home = cfg.stateDir;
+      createHome = true;
```

Comment:
You probably shouldn't use `createHome` for this. See https://github.com/NixOS/nixpkgs/pull/115332#issuecomment-792287827 for a few reasons why.

Reply:
What could be used as an alternative?

Reply:
Generally speaking, we conditionally provision system users and groups, pending that the sysadmin has not deviated from default values, like this:
https://github.com/NixOS/nixpkgs/blob/1f77a4c8c74bbe896053994836790aa9bf6dc5ba/nixos/modules/services/audio/mpd.nix#L267-L279

Doing the same for state directories is my preferred method:
https://github.com/NixOS/nixpkgs/blob/1f77a4c8c74bbe896053994836790aa9bf6dc5ba/nixos/modules/services/audio/mpd.nix#L255-L257

---
**If I were to have a concrete suggestion it would be to apply [this commit](https://github.com/NixOS/nixpkgs/commit/b2d928b7374221612c7071c4b17f93a23daa1830) to the PR. It is backwards compatible for all existing users, and clearly documents the expectations of the sysadmin if they deviate from the default value.**

Reply:
Thanks, I used your suggestion.